### ISSUE TITLE
Unable to verify S/MIME multipart/signed messages

### ISSUE THREAD

**Issue by Cerebus** [djcb/mu#1753]
## Expected or desired behavior

At a minimum, verify a correct S/MIME signed message

## Actual behavior

```
~/Maildir/gmail/INBOX/cur                                                                                                       INT
❯ openssl smime -verify -noverify < 1594331247.21315_18.penguin,U=134:2,S
Content-type: text/plain;
        charset="UTF-8"
Content-transfer-encoding: 7bit


Verification successful

~/Maildir/gmail/INBOX/cur                                                                                                       
❯ mu verify 1594331247.21315_18.penguin,U=134:2,S
Do you ultimately trust
  "CN=<deleted>"
to correctly certify user certificates?
  Yes
  Cancel
  No
[ycn]? y
Please verify that the certificate identified as:
  "CN=<deleted>"
has the fingerprint:
  D7:F1:0F:60:55:7A:16:D3:86:92:AC:98:FD:57:2C:95:0F:2A:32:43
  Correct
  Cancel
  Wrong
[ccw]? c
verdict: no signature found
```

## Steps to reproduce

For bug-reports, please describe in as much details as possible how
one can reproduce the problem*.

## Versions of mu, mu4e/emacs, operating system etc.

------------------------------------------------
mu version                           : 1.4.10

Xapian version                       : 1.4.11
GLib version                         : 2.64.3
GMime version                        : 3.2.7
Emacs version                        : GNU Emacs 26.1

Have wordexp                         : yes
Build mu4e emacs frontend            : yes
Build 'mug' toy-ui (gtk+/webkit)     : no

Have direntry->d_ino                 : yes
Have direntry->d_type                : yes
------------------------------------------------

## Any other detail

S/MIME generated by Outlook 2016/MacOS.

Yes, the `-verify -noverify` bit is a mess, but that's OpenSSL for you. `-verify` verifies the S/MIME, while `-noverify` tells it not to validate the root (which is untrusted on this machine).

**Comment by djcb** [comment]
That's not really supported using `mu` (underlying gmime only gained s/mime capabilities fairly recent); it does work in `mu4e` though, when using the gnus-based viewer.

**Comment by Cerebus** [comment]
Multiple fixes that landed in 27.1 were also required to get this all working right, but all seems good now.
